core-rust/state-manager/src/state_manager.rs

```diff
@@ -425,21 +470,49 @@ where
 
             let execute_result = series_executor.execute(&validated, "newly proposed");
             match execute_result {
-                Ok(_) => {
-                    duplicate_intent_hash_detector.record_committable_proposed(intent_hash);
-                    committable_transactions.push(CommittableTransaction {
-                        index: Some(index as u32),
-                        raw: raw_ledger_transaction,
-                        intent_hash: Some(intent_hash),
-                        notarized_transaction_hash: Some(notarized_transaction_hash),
-                        ledger_transaction_hash,
-                    });
-                    pending_transaction_results.push(PendingTransactionResult {
-                        intent_hash,
-                        notarized_transaction_hash,
-                        invalid_at_epoch,
-                        rejection_reason: None,
-                    });
+                Ok(processed_commit_result) => {
+                    match vertex_limits_tracker.check_post_execution(
+                        &self.vertex_limits_config,
+                        &processed_commit_result
+                            .local_receipt
+                            .local_execution
+                            .execution_metrics,
+                    ) {
+                        Ok(_) => {
+                            vertex_limits_tracker.advance(
```

Comment:
hmm, couldn't `.advance()` just do the "post-execution check" in its beginning, and return "yeah cool, advanced | whoops, some limit exceeded" ?

(if this seems unreasonable, then it's probably me being a psycho-fan of narrow APIs)

Reply:
I suggested this independently in the other file, where I said it was optional, but given Jakub also mentioned this, I suggest it's worth doing 👍